pkg/authorization/authorizer/interfaces.go

```diff
@@ -3,47 +3,30 @@ package authorizer
 import (
 	"net/http"
 
-	kapi "k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/apiserver/request"
-	"k8s.io/kubernetes/pkg/auth/user"
+	"k8s.io/kubernetes/pkg/auth/authorizer"
 	"k8s.io/kubernetes/pkg/util/sets"
 )
 
 type Authorizer interface {
-	Authorize(ctx kapi.Context, a Action) (allowed bool, reason string, err error)
-	GetAllowedSubjects(ctx kapi.Context, attributes Action) (sets.String, sets.String, error)
+	Authorize(a authorizer.Attributes) (allowed bool, reason string, err error)
+	GetAllowedSubjects(attributes authorizer.Attributes) (sets.String, sets.String, error)
 }
 
 type AuthorizationAttributeBuilder interface {
-	GetAttributes(request *http.Request) (Action, error)
+	GetAttributes(request *http.Request) (authorizer.Attributes, error)
 }
 
 type RequestInfoFactory interface {
 	NewRequestInfo(req *http.Request) (*request.RequestInfo, error)
 }
 
-type Action interface {
-	GetVerb() string
-	GetAPIVersion() string
-	GetAPIGroup() string
-	// GetResource returns the resource type.  If IsNonResourceURL() is true, then GetResource() is "".
-	GetResource() string
-	GetSubresource() string
-	GetResourceName() string
-	// IsNonResourceURL returns true if this is not an action performed against the resource API
-	IsNonResourceURL() bool
-	// GetURL returns the URL path being requested, including the leading '/'
-	GetURL() string
-}
-
 // ForbiddenMessageMaker creates a forbidden message from a MessageContext
 type ForbiddenMessageMaker interface {
 	MakeMessage(ctx MessageContext) (string, error)
 }
 
 // MessageContext contains sufficient information to create a forbidden message.  It is bundled in this one object to make it easy and obvious how to build a golang template
 type MessageContext struct {
```

Comment:
> Drop MessageContext and use authorizer.Attributes directly?

If you want to do it as a followup, sure.  Not needed to collapse interfaces ahead of the rebase.